File: account-service/src/main/java/com/vegesoft/moneytracker/configuration/MongoReactiveConfiguration.java
package com.vegesoft.moneytracker.configuration;

import com.mongodb.MongoClientURI;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.config.AbstractReactiveMongoConfiguration;
import org.springframework.data.mongodb.repository.config.EnableReactiveMongoRepositories;

@EnableReactiveMongoRepositories
public class MongoReactiveConfiguration extends AbstractReactiveMongoConfiguration {

    private final String mongoDbUri;

    public MongoReactiveConfiguration(@Value("${spring.data.mongodb.uri}") final String mongoDbUri) {
        this.mongoDbUri = mongoDbUri;
    }

    @Override
    public MongoClient reactiveMongoClient() {
        return MongoClients.create(clientURI().getURI());
    }

    @Override
    protected String getDatabaseName() {
        return clientURI().getDatabase();
    }

    private MongoClientURI clientURI() {
        return new MongoClientURI(mongoDbUri);
    }
}

File: statistics-service/src/main/kotlin/com/vegesoft/moneytracker/statistics/configuration/MongoReactiveConfiguration.kt
package com.vegesoft.moneytracker.statistics.configuration

import com.mongodb.MongoClientURI
import com.mongodb.reactivestreams.client.MongoClient
import com.mongodb.reactivestreams.client.MongoClients
import org.springframework.beans.factory.annotation.Value
import org.springframework.data.mongodb.config.AbstractReactiveMongoConfiguration
import org.springframework.data.mongodb.repository.config.EnableReactiveMongoRepositories

@EnableReactiveMongoRepositories
class MongoReactiveConfiguration(@Value("\${spring.data.mongodb.uri}") val mongoDbUri: String) : AbstractReactiveMongoConfiguration() {

    override fun reactiveMongoClient(): MongoClient {
        return MongoClients.create(clientURI().uri)
    }

    override fun getDatabaseName(): String {
        return clientURI().database
    }

    private fun clientURI(): MongoClientURI {
        return MongoClientURI(mongoDbUri)
    }
}